File: 20180715/Contains DuplicateII.java
ss Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer, Integer> dic = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i++){
            // Update map with new value, return old value
            Integer ord = dic.put(nums[i], i);
            if(ord != null && i - ord <= k){
                return true;
            }
        }
        return false;
    }
}

File: 20180715/Contains DuplicateII.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        dic = {}
        for i in range(len(nums)):
            if nums[i] in dic and i - dic[nums[i]] <= k:
                return True
            else:
                dic[nums[i]] = i
        return False
        
